pkg/util/parsers/parsers.go

```diff
@@ -19,37 +19,55 @@ package parsers
 import (
 	"fmt"
 
-	"github.com/docker/distribution/reference"
+	dockerref "github.com/docker/distribution/reference"
 )
 
 const (
 	defaultImageTag = "latest"
 )
 
-// parseImageName parses a docker image string into two parts: repo and tag.
-// If tag is empty, return the defaultImageTag.
-func ParseImageName(image string) (string, string, error) {
-	named, err := reference.ParseNamed(image)
+// ParseImageName parses a docker image string into three parts: repo, tag and digest.
+// If both tag and digest are empty, a default image tag will be returned.
+func ParseImageName(image string) (string, string, string, error) {
+	named, err := dockerref.ParseNamed(image)
 	if err != nil {
-		return "", "", fmt.Errorf("couldn't parse image name: %v", err)
+		return "", "", "", fmt.Errorf("couldn't parse image name: %v", err)
 	}
 
 	repoToPull := named.Name()
-	var tag string
+	var tag, digest string
 
-	tagged, ok := named.(reference.Tagged)
+	tagged, ok := named.(dockerref.Tagged)
 	if ok {
 		tag = tagged.Tag()
 	}
 
-	digested, ok := named.(reference.Digested)
+	digested, ok := named.(dockerref.Digested)
 	if ok {
-		tag = digested.Digest().String()
+		digest = digested.Digest().String()
 	}
-
 	// If no tag was specified, use the default "latest".
-	if len(tag) == 0 {
+	if len(tag) == 0 && len(digest) == 0 {
 		tag = defaultImageTag
 	}
-	return repoToPull, tag, nil
+	return repoToPull, tag, digest, nil
+}
+
+// ApplyDefaultTag parses a docker image string, if it doesn't contain any tag or digest,
+// a default tag will be applied.
+func ApplyDefaultTag(image string) (string, error) {
```

Comment:
nit: Make the function name clear s/ApplyDefaultTag/ApplyDefaultTagToImage 

Hmm...I'm not sure if this function should live in this package. Maybe we should export `defaultImageTag` and move this function to kubelet?